components/brave_vpn/brave_vpn_connection_manager_mac.mm

```diff
@@ -0,0 +1,249 @@
+  protocol_config.serverAddress = hostname;
+  protocol_config.serverCertificateCommonName = hostname;
+  protocol_config.remoteIdentifier = hostname;
+  protocol_config.enablePFS = YES;
+  protocol_config.disableMOBIKE = NO;
+  protocol_config.disconnectOnSleep = NO;
+  protocol_config.authenticationMethod =
+      NEVPNIKEAuthenticationMethodCertificate;  // to validate the server-side
+                                                // cert issued by LetsEncrypt
+  protocol_config.certificateType = NEVPNIKEv2CertificateTypeECDSA256;
+  protocol_config.useExtendedAuthentication = YES;
+  protocol_config.username = username;
+  protocol_config.passwordReference = GetPasswordRefForAccount(kBraveVPNKey);
+  protocol_config.deadPeerDetectionRate =
+      NEVPNIKEv2DeadPeerDetectionRateLow; /* increase DPD tolerance from default
+                                             10min to 30min */
+  protocol_config.useConfigurationAttributeInternalIPSubnet = false;
+
+  [[protocol_config IKESecurityAssociationParameters]
+      setEncryptionAlgorithm:NEVPNIKEv2EncryptionAlgorithmAES256];
+  [[protocol_config IKESecurityAssociationParameters]
+      setIntegrityAlgorithm:NEVPNIKEv2IntegrityAlgorithmSHA384];
+  [[protocol_config IKESecurityAssociationParameters]
+      setDiffieHellmanGroup:NEVPNIKEv2DiffieHellmanGroup20];
+  [[protocol_config IKESecurityAssociationParameters]
+      setLifetimeMinutes:1440];  // 24 hours
+  [[protocol_config childSecurityAssociationParameters]
+      setEncryptionAlgorithm:NEVPNIKEv2EncryptionAlgorithmAES256GCM];
+  [[protocol_config childSecurityAssociationParameters]
+      setDiffieHellmanGroup:NEVPNIKEv2DiffieHellmanGroup20];
+  [[protocol_config childSecurityAssociationParameters]
+      setLifetimeMinutes:480];  // 8 hours
+
+  return protocol_config;
+}
+
+}  // namespace
+
+// static
+BraveVPNConnectionManager* BraveVPNConnectionManager::GetInstance() {
+  static base::NoDestructor<BraveVPNConnectionManagerMac> s_manager;
+  return s_manager.get();
+}
+
+BraveVPNConnectionManagerMac::BraveVPNConnectionManagerMac() = default;
+BraveVPNConnectionManagerMac::~BraveVPNConnectionManagerMac() = default;
+
+void BraveVPNConnectionManagerMac::CreateVPNConnection(
+    const BraveVPNConnectionInfo& info) {
+  info_ = info;
+
+  if (StorePassword([NSString stringWithUTF8String:info_.password().c_str()],
+                    kBraveVPNKey) != errSecSuccess)
+    return;
+
+  NEVPNManager* vpn_manager = [NEVPNManager sharedManager];
+  [vpn_manager loadFromPreferencesWithCompletionHandler:^(NSError* error) {
+    if (error) {
+      LOG(ERROR) << "Create - loadFromPrefs error: "
+                 << base::SysNSStringToUTF8([error localizedDescription]);
+      return;
+    }
+
+    [vpn_manager setEnabled:YES];
+    [vpn_manager setProtocolConfiguration:CreateProtocolConfig(info_)];
+    [vpn_manager setLocalizedDescription:base::SysUTF8ToNSString(
+                                             info_.connection_name())];
```

Comment:
It would be great to add ` (<channel>)` to the connection name for ones which aren't on Release channel (ex: ` (Beta)` / ` (Dev)` / ` (Nightly)`, as you've noticed different channels can make connections on macOS with the same name.

I'm not sure which makes more sense (would leave up to you 😄 curious what other reviewers think):
- add channel here ex: 
    ```
    std::string channel_name = GetChannelNameHere();
    std::string vpn_connection_name = info_.connection_name() + " (" + channel_name + ")";
    ```
- OR
- if the channel details should be set / passed in `BraveVPNConnectionInfo` parameter (ex: implementation would set these details)

Reply:
IMO, it would be better to receive channel-aware connection name from caller because we can get channel name at chrome layer.

Reply:
SGTM! 😄